### ISSUE TITLE
Utilize user-supplied object data to populate request/assertion values

### ISSUE THREAD

**Issue by noahdietz** [apigee-127/swagger-test-templates#42]

**Comment by noahdietz** [comment]
So we are already doing this for security properties...should we consider adding this for path parameters? This might be a good next step for the issue, as they are simple data types, no potential for complex objects (i.e. in the body or formData)

**Comment by mohsen1** [comment]
What if path parameter is generated dynamically?

1. POST /pet to make a pet, grab petId
2. GET /pet/{petId} to make sure it's created
3. DELETE /pet/{petId} to remove it

**Comment by noahdietz** [comment]
Are you saying it's a reason not to implement loading from an environment variable or just saying we should consider it as a possible implementation? That sounds like it would be part of the "request series" feature we were considering. I'm thinking about all the individual tests for api calls that use the same path parameter...

**Comment by mohsen1** [comment]
I'm not saying it's a bad idea. It's actually very useful but it's only one case of many different things that people want to do.
What we're discussing here is a higher level of abstraction on top of what this module does. You can take output of this module and fiddle with the generated code (changing order of operations and filling in placeholders) to generate tests that need much less work to get it running. 
We can put that higher level of abstraction in a module that uses this module or provide a new more verbose API that does it in this module.

**Comment by noahdietz** [comment]
Right, I agree, path parameters is just a drop in the bucket of data injection, but it might be the easiest place to start?

If we built another module for test data injection, we would need to standardize the data placeholders so they are easy to manipulate post-generation. Also, the attribute that the placeholder is standing in for has to be identifiable so that it's easy to inject the data corresponding to that attribute.
This makes me wish we could generate the placeholders as Handlebars variables, i.e. ```{{userId}}```, so that it could be run through another module with an API to handle the desired data. However, such a placeholder wouldn't be valid JavaScript unless wrapped in quotes or commented out.

**Comment by noahdietz** [comment]
Could we leverage Mocha's built in [dynamic test generation](https://mochajs.org/#dynamically-generating-tests) to our advantage here? This feature is pretty cool, but might be more useful for edge-case test generation of a single path and/or redoing how we generate tests for an operation's different response codes (its the same API call, just a different input per expected response)

**Comment by mohsen1** [comment]
If I recall correctly, originally we wanted to have no test generation and do everything programmatically. With that all you needed to do was giving the tester module your swagger and some meta-data to test your API. The reason we went to code generation path was we didn't want to do stuff under the hood and only think about happy path. 
Now with test generation we can still use some of that idea, putting all request payload and other metadata like that into a config file is a good idea. But putting `it` calls in a loop puts stuff somehow under  the hood. We want users generate code and then modify it based on their specific requirements.

But still, putting things in a config file can get hairy as I commented above about path parameters. 

To be honest, I'm not sure if we should do this at all.
